packages/arb-evm/evm/merkleRootResult.go

```diff
@@ -136,9 +182,47 @@ func newMerkleTreeFromValue(val value.Value) (MerkleNode, error) {
 			return nil, err
 		}
 		return &MerkleLeaf{
-			Data: data,
+			Data:  data,
+			index: minIndex,
 		}, nil
 	} else {
 		return nil, errors.New("tree node must be a 2 or 3 tuple")
 	}
 }
+
+type MerkleRootProof struct {
+	Nodes []common.Hash
+	Path  []bool
+	Data  []byte
+}
+
+func (m *MerkleRootResult) GenerateProof(index uint64) (*MerkleRootProof, error) {
```

Comment:
We already have an existing merkle tree and proof generation system somewhere in the ethbridge MerkleTree, is that fundamentally incompatible with this in some way? I'd prefer to only have one merkle tree system in Go.

Reply:
Agreed that it'd be nice to limit these. The difference is basically that this tree is produced by ArbOS using a different structure and unifying them would require significantly altering ArbOS

Reply:
Ah, I was wondering about that. Makes sense to keep it as-is then.

Reply:
At least we use the same prover on the solidity side which minimizes the issues there and means we stick to a unified proving strategy